lib/Controller/CallController.php

```diff
@@ -0,0 +1,346 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Spreed\Controller;
+
+use OCA\Spreed\Exceptions\RoomNotFoundException;
+use OCA\Spreed\Manager;
+use OCA\Spreed\Room;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\OCSController;
+use OCP\IL10N;
+use OCP\ILogger;
+use OCP\IRequest;
+use OCP\ISession;
+use OCP\IUser;
+use OCP\IUserManager;
+
+class CallController extends OCSController {
+	/** @var string */
+	private $userId;
+	/** @var IL10N */
+	private $l10n;
+	/** @var IUserManager */
+	private $userManager;
+	/** @var ISession */
+	private $session;
+	/** @var ILogger */
+	private $logger;
+	/** @var Manager */
+	private $manager;
+
+	/**
+	 * @param string $appName
+	 * @param string $UserId
+	 * @param IRequest $request
+	 * @param IL10N $l10n
+	 * @param IUserManager $userManager
+	 * @param ISession $session
+	 * @param ILogger $logger
+	 * @param Manager $manager
+	 */
+	public function __construct($appName,
+								$UserId,
+								IRequest $request,
+								IL10N $l10n,
+								IUserManager $userManager,
+								ISession $session,
+								ILogger $logger,
+								Manager $manager) {
+		parent::__construct($appName, $request);
+		$this->userId = $UserId;
+		$this->l10n = $l10n;
+		$this->userManager = $userManager;
+		$this->session = $session;
+		$this->logger = $logger;
+		$this->manager = $manager;
+	}
+
+	/**
+	 * Get all currently existent rooms which the user has joined
+	 *
+	 * @NoAdminRequired
+	 *
+	 * @return DataResponse
+	 */
+	public function getRooms() {
+		$rooms = $this->manager->getRoomsForParticipant($this->userId);
+
+		$return = [];
+		foreach ($rooms as $room) {
+			try {
+				$return[] = $this->formatRoom($room);
+			} catch (RoomNotFoundException $e) {
+			}
+		}
+
+		return new DataResponse($return);
+	}
+
+	/**
+	 * @PublicPage
+	 *
+	 * @param string $token
+	 * @return DataResponse
+	 */
+	public function getRoom($token) {
+		try {
+			$room = $this->manager->getRoomForParticipantByToken($token, $this->userId);
+			return new DataResponse($this->formatRoom($room));
+		} catch (RoomNotFoundException $e) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
+		}
+	}
+
+	/**
+	 * @param Room $room
+	 * @return array
+	 * @throws RoomNotFoundException
+	 */
+	protected function formatRoom(Room $room) {
+		// Sort by lastPing
+		/** @var array[] $participants */
+		$participants = $room->getParticipants();
+		$sortParticipants = function(array $participant1, array $participant2) {
+			if ($participant1['lastPing'] === $participant2['lastPing']) {
```

Comment:
Hmm I think you are right. Looks like this was copied from http://php.net/manual/en/function.usort.php
And I guess they took the long way to explain it a bit better.